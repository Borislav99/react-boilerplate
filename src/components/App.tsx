import {useAppDispatch, useAppSelector} from '@hooks/index';
import {setCounter} from '@features/example/exampleSlice';
import {selectCounter} from '@features/example/selectors';

const App = () => {
  const dispatch = useAppDispatch();
  const counter = useAppSelector(selectCounter);

  const handleClick = () => {
    dispatch(setCounter(counter + 1));
  };

  return (
    <div>
      <button onClick={handleClick}>click me</button>
      <p data-testid="counter">{counter}</p>
    </div>
  );
};

export default App;
