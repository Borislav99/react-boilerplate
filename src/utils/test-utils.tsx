import React, {type PropsWithChildren} from 'react';
import {render} from '@testing-library/react';
import type {RenderOptions} from '@testing-library/react';
import {Provider} from 'react-redux';
import {storeSetup, type AppStore, type RootState} from '../store/config';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProviders(ui: React.ReactElement, extendedRenderOptions: ExtendedRenderOptions = {}) {
  const {preloadedState = {}, store = storeSetup(preloadedState), ...renderOptions} = extendedRenderOptions;

  const Wrapper = ({children}: PropsWithChildren) => <Provider store={store}>{children}</Provider>;

  return {
    store,
    ...render(ui, {wrapper: Wrapper, ...renderOptions}),
  };
}
