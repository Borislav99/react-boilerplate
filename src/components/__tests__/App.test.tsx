import {findByTestId, fireEvent, getByRole} from '@testing-library/react';
import App from '@components/App';
import {renderWithProviders} from '@utils/test-utils';

describe('App::', () => {
  it('should render app', async () => {
    const {container} = renderWithProviders(<App />);
    expect(container).toMatchSnapshot();
  });

  it('should increase counter', async () => {
    const {container} = renderWithProviders(<App />);
    const button = getByRole(container, 'button');
    fireEvent.click(button);
    fireEvent.click(button);
    expect((await findByTestId(container, 'counter')).textContent).toBe('2');
  });
});
