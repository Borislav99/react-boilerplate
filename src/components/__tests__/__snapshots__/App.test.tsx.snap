// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App:: should render app 1`] = `
<div>
  <div>
    <button>
      click me
    </button>
    <p
      data-testid="counter"
    >
      0
    </p>
  </div>
</div>
`;
